internal/server/data/access_key.go

```diff
@@ -90,6 +90,10 @@ func CreateAccessKey(db GormTxn, accessKey *models.AccessKey) (body string, err
 		accessKey.Name = fmt.Sprintf("%s-%s", identityIssuedFor.Name, accessKey.ID.String())
 	}
 
+	if err := validateAccessKey(accessKey); err != nil {
```

Comment:
Why move this to later? If the key ID is not valid, it should be rejected immediately?

Reply:
There's a bunch of things that can optionally be generated for the key before this point, so I think this spot is right for validation if want to keep it contained in one place